### PR TITLE
Moe doc fixes

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: docs/source/features/moe.rst [hunk 1]
```diff
@@ -52,16 +52,18 @@ Other options include:
 1. ``moe_input_jitter_eps`` adds noise to the input tensor by applying jitter with a specified epsilon value.
 
 2. ``moe_token_dropping`` enables selectively dropping and padding tokens for each expert to achieve
-   a specified capacity.
+   a specified capacity, similar to GShard, Switch-Transformer, and DeepSpeed-MoE. Briefly, if the number
+   of tokens routed to an expert exceed its capacity, then the exceedign tokens are dropped. Note that this is
```
Comment (ntajbakhsh) [inline docs/source/features/moe.rst:56]: exceed -> exceeds
exceedign -> exceeding

File: docs/source/features/moe.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ Mixture of Experts
 Overview
 --------
 
-NeMo supports Mixture of Experts (MoE) in the transformer layer for NLP models.
+NeMo supports Mixture of Experts (MoE) in the feedforward block of the transformer layer.
```
Comment (jgerh) [inline docs/source/features/moe.rst:7]: Add Framework.

NeMo Framework supports Mixture of Experts (MoE) in the feedforward block of the transformer layer.

File: docs/source/features/moe.rst [hunk 3]
```diff
@@ -4,7 +4,7 @@ Mixture of Experts
 Overview
```
Comment (jgerh) [inline docs/source/features/moe.rst:4]: Delete this heading here. We will move it to another location.

Overview
----------

File: docs/source/features/moe.rst [hunk 4]
```diff
@@ -4,7 +4,7 @@ Mixture of Experts
...
 --------
 
-NeMo supports Mixture of Experts (MoE) in the transformer layer for NLP models.
+NeMo supports Mixture of Experts (MoE) in the feedforward block of the transformer layer.
 
 MoE is a machine learning technique where multiple specialized models (experts,
```
Comment (jgerh) [inline docs/source/features/moe.rst:9]: Line 8. Move Overview heading here like this:

Overview
-----------

Line 14. Add a heading.

Use Moe
----------


Line 28. Add a heading.

Configure MoE-specific Loss Function
------------------------------------------

File: docs/source/features/moe.rst [hunk 5]
```diff
@@ -52,16 +52,18 @@ Other options include:
 1. ``moe_input_jitter_eps`` adds noise to the input tensor by applying jitter with a specified epsilon value.
 
 2. ``moe_token_dropping`` enables selectively dropping and padding tokens for each expert to achieve
-   a specified capacity.
+   a specified capacity, similar to GShard, Switch-Transformer, and DeepSpeed-MoE. Briefly, if the number
```
Comment (jgerh) [inline docs/source/features/moe.rst:55]: fix run-on sentence.

moe_token_dropping enables selectively dropping and padding tokens for each expert to achieve a specified capacity. Similar to GShard, Switch-Transformer, and DeepSpeed-MoE. Briefly, if the number of tokens routed to an expert exceeds its capacity, then the exceeding tokens are dropped.

File: docs/source/features/moe.rst [hunk 6]
```diff
@@ -52,16 +52,18 @@ Other options include:
...
+   of tokens routed to an expert exceeds its capacity, then the exceeding tokens are dropped. Note that this is
+    currently unsupported so should remain False.
 
-3. ``moe_token_dropping`` specifies the token dispatcher type, options include 'allgather' and 'alltoall'.
+3. ``moe_token_dispatcher_type`` specifies the token dispatcher type, options include 'allgather' and 'alltoall'.
 
 4. ``moe_per_layer_logging`` enables per-layer logging for MoE, currently support aux-loss and z-loss.
 
-5. ``moe_expert_capacity_factor`` the capacity factor for each expert, None means no token will be dropped. The default is None.
+5. ``moe_expert_capacity_factor`` the capacity factor determines the maximum number of tokens that can be routed to each expert in any MoE layer, None means no token will be dropped. The default is None.
```
Comment (jgerh) [inline docs/source/features/moe.rst:63]: fix punctuation.

moe_expert_capacity_factor the capacity factor determines the maximum number of tokens that can be routed to each expert in any MoE layer. None means no token will be dropped. The default is None.

File: docs/source/features/moe.rst [hunk 7]
```diff
@@ -52,16 +52,18 @@ Other options include:
...
 
 6. ``moe_pad_expert_input_to_capacity`` if True, pads the input for each expert to match the expert capacity length, effective only after the moe_expert_capacity_factor is set. The default setting is False.
```
Comment (jgerh) [inline docs/source/features/moe.rst:65]: fix grammar.

moe_pad_expert_input_to_capacity if True, pads the input for each expert to match the expert capacity length. It is effective only after the moe_expert_capacity_factor is set. The default setting is False.

File: docs/source/features/moe.rst [hunk 8]
```diff
@@ -52,16 +52,18 @@ Other options include:
...
 
 7. ``moe_token_drop_policy`` the policy to drop tokens. Can be either "probs" or "position". If "probs", the tokens with the lowest probabilities will be dropped. If "position", tokens at the end of each batch will be dropped. Default value is "probs".
```
Comment (jgerh) [inline docs/source/features/moe.rst:67]: Add "the".

moe_token_drop_policy the policy to drop tokens. Can be either "probs" or "position". If "probs", the tokens with the lowest probabilities will be dropped. If "position", tokens at the end of each batch will be dropped. The default value is "probs".

File: docs/source/features/moe.rst [hunk 9]
```diff
@@ -52,16 +52,18 @@ Other options include:
...
 
-8. ``moe_layer_recompute`` if True, checkpointing moe_layer to save activation memory, default is False.
\ No newline at end of file
+8. ``moe_layer_recompute`` if True, checkpointing moe_layer to save activation memory, default is False.
```
Comment (jgerh) [inline docs/source/features/moe.rst:69]: fix sentence.

``moe_layer_recompute`` if True, checkpointing moe_layer to save activation memory. The default is False.

**Review by jgerh — COMMENTED** [review]
Completed the review again of the moe.rst file. Please add all copyedits for grammar, punctuation, and NeMo Framework naming conventions. Also, make the requested heading changes for Overview, Use MoE, and Configure MoE-specific Loss Function.

**Review by jgerh — APPROVED** [review]
Approved changes.

**Review by thomasdhc — APPROVED** [review]

**Pull request merged**